nemo/collections/llm/bert/loss.py

```diff
@@ -241,3 +238,16 @@ def sentence_order_prediction_loss(tensor: Tensor, sentence_order: Tensor):
     loss = F.cross_entropy(losses, sentence_order, ignore_index=-1)
 
     return loss
+
+
+def average_losses_across_data_parallel_group(losses):
+    """Reduce a tensor of losses across all GPUs."""
+    from megatron.core import parallel_state
+
+    averaged_losses = torch.cat([loss.clone().detach().view(1) for loss in losses])
+    torch.distributed.all_reduce(averaged_losses, group=parallel_state.get_data_parallel_group())
+    averaged_losses = averaged_losses / torch.distributed.get_world_size(
+        group=parallel_state.get_data_parallel_group()
+    )
+
+    return averaged_losses
```

Comment:
instead of copy/pasting, does it make sense to move this from `nemo.collections.nlp.modules.common.megatron.utils` to somewhere like here? https://github.com/NVIDIA/NeMo/blob/main/nemo/utils/distributed.py

and have nemo1 and nemo2 use the same function then?

Reply:
> instead of copy/pasting, does it make sense to move this from `nemo.collections.nlp.modules.common.megatron.utils` to somewhere like here? https://github.com/NVIDIA/NeMo/blob/main/nemo/utils/distributed.py
> 
> and have nemo1 and nemo2 use the same function then?

@ananthsub looks like this function is only used for bert. Also, we are going to deprecate nemo1 soon so we don't need to have this function for nemo1 in future